pkg/ccl/backupccl/backup_processor.go

```diff
@@ -312,23 +317,71 @@ func runBackupProcessor(
 	clusterSettings := flowCtx.Cfg.Settings
 
 	totalSpans := len(spec.Spans) + len(spec.IntroducedSpans)
-	todo := make(chan spanAndTime, totalSpans)
-	var spanIdx int
-	for _, s := range spec.IntroducedSpans {
-		todo <- spanAndTime{
-			spanIdx: spanIdx, span: s, firstKeyTS: hlc.Timestamp{}, start: hlc.Timestamp{},
-			end: spec.BackupStartTime,
+	requestSpans := make([]spanAndTime, 0, totalSpans)
+	rangeSizedSpans := preSplitExports.Get(&flowCtx.EvalCtx.Settings.SV)
+
+	for _, fullSpan := range spec.IntroducedSpans {
+		remainingSpan := fullSpan
+
+		if rangeSizedSpans {
+			rdi, err := flowCtx.Cfg.ExecutorConfig.(*sql.ExecutorConfig).RangeDescIteratorFactory.NewIterator(ctx, fullSpan)
+			if err != nil {
+				return err
+			}
+			for ; rdi.Valid(); rdi.Next() {
+				rangeDesc := rdi.CurRangeDescriptor()
+				subspan := remainingSpan.Intersect(roachpb.Span{Key: rangeDesc.StartKey.AsRawKey(), EndKey: rangeDesc.EndKey.AsRawKey()})
+
+				requestSpans = append(requestSpans, spanAndTime{
+					span: subspan, firstKeyTS: hlc.Timestamp{}, start: hlc.Timestamp{},
+					end: spec.BackupStartTime,
+				})
+				remainingSpan.Key = subspan.EndKey
+			}
+
+			if !remainingSpan.Overlaps(fullSpan) {
+				continue
+			}
 		}
-		spanIdx++
+
+		requestSpans = append(requestSpans, spanAndTime{
+			span: remainingSpan, firstKeyTS: hlc.Timestamp{}, start: hlc.Timestamp{},
+			end: spec.BackupStartTime,
+		})
 	}
-	for _, s := range spec.Spans {
-		todo <- spanAndTime{
-			spanIdx: spanIdx, span: s, firstKeyTS: hlc.Timestamp{}, start: spec.BackupStartTime,
-			end: spec.BackupEndTime,
+
+	for _, fullSpan := range spec.Spans {
+		remainingSpan := fullSpan
+
+		if rangeSizedSpans {
+			rdi, err := flowCtx.Cfg.ExecutorConfig.(*sql.ExecutorConfig).RangeDescIteratorFactory.NewIterator(ctx, fullSpan)
```

Comment:
I could see breaking out a function for splitting the spans. Although we'll want to be careful to not end up allocating/copying more than we need to as a result.

Reply:
Eh, the spans they put in the slice are different, so the reused function would really just be driving an iterator, and driving an iterator doesn't seem worth reusing to me.

Reply:
Actually, I tried it and it is cleaner. Done.